audio/writer.h

```diff
@@ -0,0 +1,26 @@
+#ifndef AUDIO_WRITER_H
+#define AUDIO_WRITER_H
+
+#include <stdio.h>
+#include <soundio/soundio.h>
+
+struct audio_writer {
```

Comment:
Elsewhere, you are embedding `struct audio_writer` into the per-writer structures. I would rather have `struct audio_writer` contain an opaque pointer to the context of the thing it is using. This does mean an additional allocation, but it gets rid of `container_of`, which I don't like very much. The other thing is that it makes it easier for us to transition `struct audio_writer` to something like a `struct audio_module`, which I think would be a nice way to move with this interface (as mentioned above).